engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/optimize/OptimizeCompletedHistoricIncidentRestServiceTest.java

```diff
@@ -0,0 +1,166 @@
+  }
+
+  @Test
+  public void testFinishedAfterQueryParameter() {
+    Date now = new Date();
+    given()
+      .queryParam("finishedAfter", DATE_FORMAT_WITH_TIMEZONE.format(now))
+    .then()
+      .expect()
+        .statusCode(Status.OK.getStatusCode())
+        .contentType(MediaType.APPLICATION_JSON)
+    .when()
+      .get(OPTIMIZE_COMPLETED_HISTORIC_INCIDENT_PATH);
+
+    verify(mockedOptimizeService).getCompletedHistoricIncidents(now, null, Integer.MAX_VALUE);
+    verifyNoMoreInteractions(mockedOptimizeService);
+  }
+
+  @Test
+  public void testFinishedAtQueryParameter() {
+    Date now = new Date();
+    given()
+      .queryParam("finishedAt", DATE_FORMAT_WITH_TIMEZONE.format(now))
+    .then()
+      .expect()
+        .statusCode(Status.OK.getStatusCode())
+        .contentType(MediaType.APPLICATION_JSON)
+    .when()
+      .get(OPTIMIZE_COMPLETED_HISTORIC_INCIDENT_PATH);
+
+    verify(mockedOptimizeService).getCompletedHistoricIncidents(null, now, Integer.MAX_VALUE);
+    verifyNoMoreInteractions(mockedOptimizeService);
+  }
+
+  @Test
+  public void testMaxResultsQueryParameter() {
+    given()
+      .queryParam("maxResults", 10)
+    .then()
+      .expect()
+        .statusCode(Status.OK.getStatusCode())
+        .contentType(MediaType.APPLICATION_JSON)
+    .when()
+      .get(OPTIMIZE_COMPLETED_HISTORIC_INCIDENT_PATH);
+
+    verify(mockedOptimizeService).getCompletedHistoricIncidents(null, null, 10);
+    verifyNoMoreInteractions(mockedOptimizeService);
+  }
+
+  @Test
+  public void testQueryParameterCombination() {
+    Date now = new Date();
+    given()
+      .queryParam("finishedAfter", DATE_FORMAT_WITH_TIMEZONE.format(now))
+      .queryParam("finishedAt", DATE_FORMAT_WITH_TIMEZONE.format(now))
+      .queryParam("maxResults", 10)
+    .then()
+      .expect()
+        .statusCode(Status.OK.getStatusCode())
+        .contentType(MediaType.APPLICATION_JSON)
+    .when()
+      .get(OPTIMIZE_COMPLETED_HISTORIC_INCIDENT_PATH);
+
+    verify(mockedOptimizeService).getCompletedHistoricIncidents(now, now, 10);
+    verifyNoMoreInteractions(mockedOptimizeService);
+  }
+
+  @Test
+  public void testPresenceOfProcessInstanceIdProperty() {
+    final HistoricIncidentEntity mock = mock(HistoricIncidentEntity.class);
+    when(mock.getProcessInstanceId()).thenReturn(MockProvider.EXAMPLE_PROCESS_INSTANCE_ID);
+    when(mockedOptimizeService.getCompletedHistoricIncidents(null, null, Integer.MAX_VALUE))
+      .thenReturn(Collections.singletonList(mock));
+
+    final Response response = given()
+      .then()
+        .expect()
+          .statusCode(Status.OK.getStatusCode())
+          .contentType(MediaType.APPLICATION_JSON)
+      .when()
+        .get(OPTIMIZE_COMPLETED_HISTORIC_INCIDENT_PATH);
+
+    String content = response.asString();
+    String processInstanceId = from(content).getString("[0].processInstanceId");
+
+    assertEquals(MockProvider.EXAMPLE_PROCESS_INSTANCE_ID, processInstanceId);
```

Comment:
Good idea, but I'm not sure if it adds a lot of value. I think this would just test the mapping library. I'll keep it as it is if you don't have any profound reasons to test all the fields.

Reply:
It also tests the presence of the properties in the DTO. But I'm fine either way.

Reply:
I gave it a try and wanted to reuse the mock methods that are already used by other test classes. However, it turns out to be more complicated so I'll leave it as it is as the value isn't worth the effort imho.